include/sound/sof/topology.h

```diff
@@ -75,11 +75,20 @@ struct sof_ipc_comp {
 #define SOF_MEM_CAPS_CACHE			(1 << 6) /**< cacheable */
 #define SOF_MEM_CAPS_EXEC			(1 << 7) /**< executable */
 
+#define SOF_BUF_FLAG_OVERRUN_NO_XRUN	(1 << 0) /**< overrun will cause ring
+						  * buffer overwrite, instead
+						  * of XRUN */
+#define SOF_BUF_FLAG_UNDERRUN_NO_XRUN	(1 << 1) /**< underrun will cause
+						  * readback of 0s, instead of
+						  * XRUN */
+
 /* create new component buffer - SOF_IPC_TPLG_BUFFER_NEW */
 struct sof_ipc_buffer {
 	struct sof_ipc_comp comp;
 	uint32_t size;		/**< buffer size in bytes */
 	uint32_t caps;		/**< SOF_MEM_CAPS_ */
+	uint32_t flags;		/**< SOF_BUF_FLAG_ */
+	uint32_t reserved;	/**< reserved for future use */
```

Comment:
isn't this going to break backwards compatibility?

Reply:
@plbossart AFAIK, this structure always sent standalone, so payload mixing shouldn't occur at any place.
As for default behaviour, flags are defined in a way, that zero-value is the current default XRUN behaviour, while setting these flags alters it.
So both ways, this should be backwards compatible.

Reply:
@akloniex how about mixing versions of firmware and kernel - one supporting these fields, another one not supporting them? Wouldn't the firmware at least complain about a wrong size and fail? Wouldn't at least the ABI version have to be incremented?

Reply:
@akloniex FW will need to deal with old/new versions of this struct (based on size).

Reply:
I think minor ABI bump is needed so that host can know whether the FW it is talking to honors these flags or not. And FW needs to handle correctly the case of an older host (but that seems to be thought out already).
@akloniex Can you please add a note summarizing how ABI compatibility is covered (i.e. summary of conclusions of this discussion)

Reply:
@kv2019i ABI compatibility is handled seamlessly, due to the nature of the change and future-planning on FW side in IPC handler.
FW, based on payload size reported by kernel, will distinguish between newer (bigger) and older structs, and in case of receiving mismatching size, would zero-pad fields not covered by received payload (older kernel) or read only part that it can parse (old FW, new kernel).
This way we have both-ways backward-compatible change.
If you wish to have that written down somewhere in the code/commit message, please tell me where to put such note.

Reply:
Typo in commit message: permmitted

You can add a paragraph in the commit message based on your explanations, that's good enough

Reply:
@akloniex Thanks, the commit message is very informative now. Also no need to explain the general ABI extension mechanism in the commit message. You could add one sentence to second paragraph: "This is an incremental ABI change as the new fields are ignored by older versions of the firmware."

Reply:
@plbossart @kv2019i done as requested.

Reply:
@akloniex looks good now, thanks!